src/lightning/fabric/strategies/fsdp.py

```diff
@@ -895,8 +808,15 @@ def _load_raw_module_state(state_dict: Dict[str, Any], module: Module, world_siz
     if not isinstance(module, FSDP):
         module.load_state_dict(state_dict, strict=strict)
     else:
-        with _get_full_state_dict_context(module, world_size=world_size, rank0_only=False):
-            module.load_state_dict(state_dict, strict=strict)
+        if _TORCH_GREATER_EQUAL_2_2:
+            from torch.distributed.checkpoint.state_dict import StateDictOptions, set_model_state_dict
+
+            # `cpu_offload` disabled because when used with `full_state_dict` only rank 0 loads the state dict
```

Comment:
Notice that the other path sets `rank0_only=False`. I asked if this could be configurable in https://github.com/pytorch/pytorch/pull/112837#discussion_r1494626221